bolt_unix.go

```diff
@@ -12,6 +12,59 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// pwritev writes the given pages using vectorized io. The pages must be sorted by their id indicating their sequence.
+// The logic identifies runs of consecutive pages that are written with vectorized io using the pwritev syscall.
+func pwritev(db *DB, pages []*page) error {
+	offsets, iovecs := pagesToIovec(db, pages)
+
+	var (
+		e syscall.Errno
+	)
+	for i := 0; i < len(offsets); i++ {
+		_, _, e = syscall.Syscall6(syscall.SYS_PWRITEV, db.file.Fd(), uintptr(unsafe.Pointer(&iovecs[i][0])),
+			uintptr(len(iovecs[i])), uintptr(offsets[i]), uintptr(offsets[i]>>0x8), 0)
```

Comment:
The syscall pacakge is deprecated and locked down. Please consider using `golang.org/x/sys` instead. I had a quick look at your another solution [vectorizedio2](https://github.com/etcd-io/bbolt/compare/master...tjungblu:bbolt:vectorizedio2), it seems that you did not cast `iovecs` to [][]byte correctly (about line 556 in tx.go).

Reply:
yeah, let me try again with the other solution. I'm afraid this will be drastically slower due to the extra allocation however.

Reply:
yep, it's significantly slower:

```
with x/sys:
[tjungblu ~/git/bbolt]$ go run cmd/bbolt/main.go bench -count 100000
# Write	199.840821ms	(1.998µs/op)	(500500 op/sec)
# Read	1.000347769s	(17ns/op)	(58823529 op/sec)


this PR:
[tjungblu ~/git/bbolt]$ go run cmd/bbolt/main.go bench -count 100000
# Write	173.741269ms	(1.737µs/op)	(575705 op/sec)
# Read	1.001078252s	(14ns/op)	(71428571 op/sec)


master:
[tjungblu ~/git/bbolt]$ go run cmd/bbolt/main.go bench -count 100000
# Write	184.005891ms	(1.84µs/op)	(543478 op/sec)
# Read	1.000812791s	(14ns/op)	(71428571 op/sec)

```

Reply:
pushed the code into another branch https://github.com/etcd-io/bbolt/compare/master...tjungblu:bbolt:vec2